executor/split.go

```diff
@@ -74,28 +106,40 @@ func (e *SplitIndexRegionExec) Next(ctx context.Context, _ *chunk.Chunk) error {
 				zap.Error(err))
 			continue
 		}
+		if regionID == 0 {
+			continue
+		}
 		regionIDs = append(regionIDs, regionID)
 
-		if isCtxDone(ctxWithTimeout) {
-			return errors.Errorf("wait split region timeout(%v)", e.ctx.GetSessionVars().GetSplitRegionTimeout())
-		}
 	}
+	e.splitRegions = len(regionIDs)
 	if !e.ctx.GetSessionVars().WaitSplitRegionFinish {
 		return nil
 	}
+	remainMillisecond := 0
+	finishScatterNum := 0
 	for _, regionID := range regionIDs {
-		err := s.WaitScatterRegionFinish(regionID)
-		if err != nil {
+		if isCtxDone(ctxWithTimeout) {
+			// Do not break here for checking remain region scatter finished with a very short back off time.
+			// Imagine this situation, we split region 1,2,3, and timeout on wait region 1 scatter,
```

Comment:
IMHO the comment is still not good enough... would you please get help from some other colleagues, like i18n team?

Reply:
done.